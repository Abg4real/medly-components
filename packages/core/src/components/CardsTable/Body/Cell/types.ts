import { ColumnConfig } from '../../types';

export interface Props {
    data: any;
    column: ColumnConfig;
    rowId: any;
    isRowClickDisabled?: boolean;
}

export interface StyledProps {
    flex: number;
    align: 'left' | 'center' | 'right';
}
