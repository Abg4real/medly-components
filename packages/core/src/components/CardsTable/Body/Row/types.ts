import { WithThemeProp } from '@medly-components/utils';

export interface Props extends WithThemeProp {
    onClick?: () => void;
    withWhiteBackground: boolean;
}
