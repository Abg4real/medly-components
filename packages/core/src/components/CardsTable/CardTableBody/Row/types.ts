import { Props as CardProps } from '../../../Card/types';

export interface Props extends CardProps {
    disabled?: boolean;
    withLightTheme: boolean;
}
