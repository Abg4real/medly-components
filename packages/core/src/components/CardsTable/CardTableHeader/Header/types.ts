import { SortOrder } from '../../types';
import { WithThemeProp } from '@medly-components/utils/src';

export interface Props extends WithThemeProp{
    sortField?: string;
    field?: string;
    onSortChange?: (field: string, order: SortOrder) => void;
    enableSorting?: boolean;
    title: string;
}    