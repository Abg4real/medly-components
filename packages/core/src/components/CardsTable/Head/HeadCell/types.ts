import { WithThemeProp } from '@medly-components/utils/src';
import { ColumnConfig, SortOrder } from '../../types';

export interface Props extends WithThemeProp {
    column: ColumnConfig;
    sortField?: string;
    onSortChange?: (field: string, order: SortOrder) => void;
}

export interface StyledProps {
    flex: number;
    align: 'left' | 'center' | 'right';
}
