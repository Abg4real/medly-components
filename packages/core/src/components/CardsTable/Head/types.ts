import { ColumnConfig, SortOrder } from '../types';

export interface Props {
    columns: ColumnConfig[];
    onSort?: (field: string, order: SortOrder) => void;
}
