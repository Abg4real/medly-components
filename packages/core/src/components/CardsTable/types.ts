import { HTMLProps } from '@medly-components/utils';
import ColumnConfiguration from '../Table/ColumnConfiguration';

export type ItemsPosition = 'top' | 'bottom' | 'left' | 'right' | 'center';
export type Variants = 'outlined' | 'solid' | 'flat';
export type FlowDirection = 'vertical' | 'horizontal';
export type SortOrder = 'asc' | 'desc';

export interface Data {
    [key: string]: any;
}

export interface ColumnsWidth {
    numeric: string;
    group: string;
    checkbox: string;
    boolean: string;
    'text-short': string;
    'text-long': string;
    'react-component': string;
}

export interface ColumnConfig {
    /** To show sort icons */
    sort?: boolean;
    /** To hide the column */
    title: string;
    /** Field name in the data */
    field: string;
    /** Column content alignment */
    align?: 'left' | 'right' | 'center';
    /** Column flex */
    flex?:number
    /** Custom component */
    component?: React.SFC<{ data: any; rowId?: any; disabled?: boolean }>;
}
export interface Props extends Omit<HTMLProps<HTMLOListElement>, 'data' | 'type'> {
    /** Table data */
    data: Data[];
    /** Key name in data to be used as unique id */
    uniqueKeyName?: string;
    /** Column configuration */
    columns: ColumnConfig[];
    /** Shows placeholder shimmer */
    isLoading?: boolean;
    /** Function to be called on click of sort icon */
    onSort?: (field: string, order: SortOrder) => void;
    /** Function to be called on row click */
    onRowClick?: (rowData: object) => void;
    /** Key name to disable row click */
    rowClickDisableKey?: string;
    /** Key name to select background color */
    withLightTheme?: boolean;

}
export interface StaticProps {
    ColumnConfiguration?: typeof ColumnConfiguration;
}
