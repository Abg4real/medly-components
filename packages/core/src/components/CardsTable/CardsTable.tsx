import React, { SFC } from 'react';
import Body from './Body';
import * as Styled from './CardTable.styled';
import Head from './Head';
import { Props } from './types';

export const CardsTable: SFC<Props> = React.memo(({ withWhiteBackground, onSort, columns, ...restProps }) => (
    <Styled.Table>
        <Head onSort={onSort} columns={columns} withWhiteBackground={withWhiteBackground} />
        <Body {...restProps} columns={columns} withWhiteBackground={withWhiteBackground} />
    </Styled.Table>
));

CardsTable.displayName = 'CardsTable';
CardsTable.defaultProps = {
    withWhiteBackground: false
};
