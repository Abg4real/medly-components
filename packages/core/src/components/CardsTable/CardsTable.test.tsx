import { render } from '@test-utils';
import React from 'react';
import { CardsTable } from './CardsTable'
import { ColumnConfig } from './types';

describe('CardsTable', () => {
    it('should render CardsTable of cards table properly', () => {

        // const { data, onSort, columns, onRowClick, rowClickDisableKey, uniqueKeyName, ...restProps } = props; 
        
        const onSort = jest.fn(),
            onRowClick = jest.fn(),
            columns: ColumnConfig[] = [
            { title: 'Patient Info', field: 'patientInfo', formatter: 'text-short', sort: true },
            { title: 'RxInfo', field: 'rxInfo', formatter: 'group', sort: true }
        ];
        const data = [{
            id: 1,
            patientInfo: 'Oli Bob',
            rxInfo: 'Metaformin',
            price: '535.76',
            status: 'Pending'
            
        }];
        
        const { container } = render(<CardsTable 
            data={data}
            onSort={onSort}
            columns={columns}
            onRowClick={onRowClick}
            rowClickDisableKey='disable'
            uniqueKeyName='123'

         />);
        expect(container).toMatchSnapshot();
    });
});
