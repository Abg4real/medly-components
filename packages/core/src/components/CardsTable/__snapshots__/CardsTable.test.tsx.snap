// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsTable should render CardsTable properly 1`] = `
.c4 {
  margin: 0;
  color: inherit;
  font-size: 1.2rem;
  font-weight: 700;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
}

.c8 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
}

.c7 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0;
  margin: 0 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  cursor: pointer;
  box-sizing: border-box;
  padding: 1.6rem 0;
  border-radius: 0.8rem;
  margin-bottom: 0.8rem;
  background-color: white;
  border: 0.1rem solid #ffffff;
  box-shadow: 0 0.2rem 0.8rem rgba(176,188,200,0.2);
}

.c6:hover {
  box-shadow: 0 0.4rem 0.8rem rgba(176,188,200,0.6);
}

.c6:nth-child(odd) {
  background: #ffffff;
}

.c6:nth-child(even) {
  background: #ffffff;
}

.c0 {
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background-color: transparent;
  margin: 2.4rem 0;
}

.c5 {
  font-size: 1.6rem;
  cursor: pointer;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
}

.c5 * {
  fill-opacity: 1;
  fill: #607890;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0;
  margin: 0 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c2 .c3 {
  display: inline;
  color: #435465;
}

.c2 .sc-AykKI {
  cursor: pointer;
  padding-left: 1rem;
}

.c2 .sc-AykKI * {
  fill: #435465;
}

<div>
  <table
    class="c0"
  >
    <thead>
      <tr
        class="c1"
      >
        <th
          class="c2"
        >
          <h5
            class="c3 c4"
          >
            Patient Info
          </h5>
          <svg
            class="c5"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.208 5.03a1 1 0 0 1 1.584 0l3.165 4.11a1 1 0 0 1-.793 1.61H8.836a1 1 0 0 1-.793-1.61l3.165-4.11zM12.792 18.97a1 1 0 0 1-1.584 0l-3.165-4.11a1 1 0 0 1 .793-1.61h6.328a1 1 0 0 1 .793 1.61l-3.165 4.11z"
              fill="#000"
            />
          </svg>
        </th>
        <th
          class="c2"
        >
          <h5
            class="c3 c4"
          >
            RxInfo
          </h5>
          <svg
            class="c5"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.208 5.03a1 1 0 0 1 1.584 0l3.165 4.11a1 1 0 0 1-.793 1.61H8.836a1 1 0 0 1-.793-1.61l3.165-4.11zM12.792 18.97a1 1 0 0 1-1.584 0l-3.165-4.11a1 1 0 0 1 .793-1.61h6.328a1 1 0 0 1 .793 1.61l-3.165 4.11z"
              fill="#000"
            />
          </svg>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="c6"
      >
        <td
          class="c7"
        >
          <span
            class="c3 c8"
          >
            Oli Bob
          </span>
        </td>
        <td
          class="c7"
        >
          <span
            class="c3 c8"
          >
            Metaformin
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
